pennylane/workflow/qnode.py

```diff
@@ -110,15 +110,38 @@ def _to_qfunc_output_type(
     if has_partitioned_shots:
         return tuple(_to_qfunc_output_type(r, qfunc_output, False) for r in results)
 
-    # Special case of single Measurement in a list
-    if isinstance(qfunc_output, list) and len(qfunc_output) == 1:
+    qfunc_output_leaves, qfunc_output_structure = qml.pytrees.flatten(
+        qfunc_output, is_leaf=lambda obj: isinstance(obj, (qml.measurements.MeasurementProcess))
+    )
+    results_leaves, _ = qml.pytrees.flatten(results)
+
+    if len(results_leaves) != len(qfunc_output_leaves) or any(
+        [
+            qml.typing._is_jax(results),
+            qml.typing._is_torch(results),
+            qml.typing._is_tensorflow(results),
+        ]
+    ):
+
+        # # Special case of single Measurement in a list
+        if isinstance(qfunc_output, list) and len(qfunc_output) == 1:
+            results = [results]
+
+        # If the return type is not tuple (list or ndarray) (Autograd and TF backprop removed)
+        if isinstance(qfunc_output, (tuple, qml.measurements.MeasurementProcess)):
+            return results

```

Comment:
```suggestion
        if isinstance(qfunc_output, (list, tuple, qml.measurements.MeasurementProcess)):
            return results
```

Same comment as above. Anything we do especially for either list or tuple, we should do for the other type.